src/testFixtures/java/uk/gov/di/ipv/cri/common/library/client/CommonApiClient.java

```diff
@@ -88,6 +105,60 @@ public HttpResponse<String> sendNewSessionRequest(String sessionRequestBody)
         return sendHttpRequest(request);
     }
 
+    public HttpResponse<String> sendEventRequest(String sessionId)
+            throws IOException, InterruptedException {
+
+        URI eventsEndpointURI =
+                URI.create(
+                        "https://"
+                                + STACK_NAME
+                                + ".review-a."
+                                + ENV
+                                + ".account.gov.uk/events?partitionKey=SESSION%23"
+                                + sessionId
+                                + "&sortKey=TXMA");
+
+        return sendRequest(
+                signRequest(
+                        SdkHttpFullRequest.builder()
+                                .method(SdkHttpMethod.GET)
+                                .uri(eventsEndpointURI)
+                                .build(),
+                        AwsSessionCredentials.create(
+                                AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, AWS_SESSION_TOKEN)));
+    }
+
+    private SdkHttpRequest signRequest(
+            SdkHttpFullRequest unsignedRequest, AwsSessionCredentials credentials) {
+        AwsV4HttpSigner signer = AwsV4HttpSigner.create();
+        SignedRequest signedRequest =
+                signer.sign(
+                        r ->
+                                r.identity(credentials)
+                                        .request(unsignedRequest)
+                                        .putProperty(
+                                                AwsV4HttpSigner.SERVICE_SIGNING_NAME, AWS_SERVICE)
+                                        .putProperty(AwsV4HttpSigner.REGION_NAME, REGION.id()));
+
+        return signedRequest.request();
+    }
+
+    private HttpResponse<String> sendRequest(SdkHttpRequest sdkHttpRequest)
+            throws IOException, InterruptedException {
+        HttpRequest.Builder requestBuilder = HttpRequest.newBuilder();
+        requestBuilder.GET();
+        requestBuilder.uri(sdkHttpRequest.getUri());
+        for (Map.Entry<String, List<String>> entry : sdkHttpRequest.headers().entrySet()) {
```

Comment:
Not essential, but you could look into what lines 151-158 would look like use `stream()`

It may be quite complex and less readable than what is currently there though so only if it's better.

Reply:
I was looking at examples of using `stream()` but it seems like I would still have to `.flatMap()` and still use `Map.Entry` in the stream suggestions

Reply:
- +1 to using stream here, it should work quite well